fluent/client/ws/connection.go

```diff
@@ -111,16 +118,21 @@ func (wsc *connection) CloseWithMsg(closeCode int, msg string) error {
 	return nil
 }
 
-func (wsc *connection) enqueueErr(err error) {
-	wsc.closedLock.Lock()
-	defer wsc.closedLock.Unlock()
+func (wsc *connection) Close() error {
```

Comment:
@BernhardSchmid Yeah... I think I introduced a bug here because `Close` should wait for the read loop to close or timeout. You can see the pattern in this example.

https://github.com/gorilla/websocket/blob/master/examples/echo/client.go#L40-L81

I'll fix it

Reply:
I remember thinking "Why did I make it so complicated before?". Now I know ;)